src/System.Collections.Specialized/src/System/Collections/Specialized/OrderedDictionary.cs

```diff
@@ -253,8 +266,10 @@ public void Clear()
             {
                 throw new NotSupportedException(SR.OrderedDictionary_ReadOnly);
             }
-            objectsTable.Clear();
-            objectsArray.Clear();
+            EnsureObjectsTable();
```

Comment:
If `Count == 0` then you can return without initializing anything.

Reply:
Internally Clear checks Count/Size on first step so I only replaced EnsureObjectsTable/EnsureObjectsArray calls with null checks.